modules/ROOT/pages/_partials/data-sync/js-client-conflict.adoc

```diff
@@ -1,84 +1,184 @@
 
-Developers can supply their own `conflictListener` implementation to get notifications about conflicts:
+{sync-service} allows developers to receive information about the data conflict that occurred between the client and the server.
+
+When a conflict occurs {sync-service} will attempt to do a field level resolution of data - meaning it will check all fields of your type to see if both the client or server has changed the same field. The client can be notified in one of two scenarios.
+
+If both client and server have changed any of the same fields then the `conflictOccurred` method of your `ConflictListener` will be triggered.
+
+If the client and server have not changed any of the same fields and the data can be easily merged then the `mergeOccurred` method of your `ConflictListener` will be triggered.
+
+Developers can supply their own `conflictListener` implementation
 
 [source,javascript]
 ----
+class ConflictLogger implements ConflictListener {
+  conflictOccurred(operationName, resolvedData, server, client) {
+    console.log("Conflict occurred with the following:")
+    console.log(`data: ${JSON.stringify(resolvedData)}, server: ${JSON.stringify(server)}, client: ${JSON.stringify(client)}, operation:  ${JSON.stringify(operationName)}`);
+  }
+  mergeOccurred(operationName, resolvedData, server, client) {
+    console.log("Merge occurred with the following:")
+    console.log(`data: ${JSON.stringify(resolvedData)}, server: ${JSON.stringify(server)}, client: ${JSON.stringify(client)}, operation:  ${JSON.stringify(operationName)}`);
+  }
+}
+
 let config = {
 ...
-  conflictListener: {
-    conflictOccurred: function(operationName, resolvedData, server, client) {
-      console.log(`data: ${JSON.stringify(resolvedData)}, server: ${JSON.stringify(server)} client: ${JSON.stringify(client)} `);
-    }
-  }
+  conflictListener: new ConflictLogger()
 ...
 }
 ----
+
+== Pre-Conflict Errors
+
+{sync-service} provides a mechanism for developers to check for a 'pre-conflict' before a mutation occurs. It does this out of the box by checking whether or not the data being sent conflicts locally. This happens when a mutation (or the act of creating a mutation) is initiated and before being sent new data arrives via subscriptions.
+
+An example of when this is useful could be when a user performs the following actions:
+
+. Open a form on their device
+. Begin working on the pre-populated data on this form
+. Whilst working the client receives new data from the server via subscriptions
+. The client is now conflicted but the user is unaware.
+. When the user presses submit {sync-service} notices that their data is conflicted and provides the developer with a way to warn the user.
+
+To use this feature, and therefore potentially save unecessary round-trips to the server with data which is definitely conflicted, developers can make use of the error returned by {sync-service}. An example of how developers can use this error can be seen below.
```

Comment:
I think this is more like about not letting this change to be enqueued for offline when we 100% know that this is going to be conflicted.

Reply:
This is also a valid point though